### ISSUE TITLE
The URL of a post is overwrote to http://0.0.7.225/10/07/Jekyll-blog/

### ISSUE THREAD

**Issue by binlidaily** [poole/lanyon#209]
When I click one post from http://127.0.0.1:4000/, the URL of the post is overwrote to http://0.0.7.225/10/07/Jekyll-blog/. Of course, it's unreachable. Do you know what I can do for this issue?

I just copy the lanyon source code to the folder of my github page, the problem is there.

**Comment by rajeevkannav** [comment]
Try to match your _config.yml with _config.yml of gh-pages branch.

**Comment by johnta0** [comment]
@binlidaily I got the same problem.

**Comment by mjthompson** [comment]
I first tried to fix this by changing baseurl in _config.yml, but it caused every page to 404.

Try changing references to {{site.baseurl}} to {{site.url}} instead.

**Comment by ManZzup** [comment]
This happens to due a slight bug in index.html
https://github.com/poole/lanyon/blob/ac85a3115c094b3e6cb92cc41771e98b443f6c5d/index.html#L10-L12
 
I had the same bug and fixed it by changing
```
 <a href="{{ site.baseurl }}/{{ post.url }}">
```

to 

```
 <a href="{{ site.baseurl }}{{ post.url }}">
```

How you get the above IP is when you have `base-url: "/"`, GitHub pages seems to overwrite the url to "" as well. Then the post links becomes in the format [//2017/10/07/Jekyll-blog/](//2017/10/07/Jekyll-blog/). Now the browser treats this as an address and try to decode the first number into an IP giving **0.0.7.225**

I assume that this bug started after Jekyll decided to remove "/" from the baseurl and added into post.url, anyhow, above fix worked for me so far.

**Comment by mdo** [comment]
Fixed by #238.
